src/utils.js

```diff
@@ -0,0 +1,57 @@
+//Groups array of objects by a property.
+export const groupBy = (arr, property) => {
+  return arr.reduce((r, a) => {
+    r[a[property]] = [...(r[a[property]] || []), a]
+    return r
+  }, {})
+}
+
+//Prototype for filter object data by a predicate.
+Object.filter = (obj, predicate) =>
+  Object.keys(obj)
+    .filter(key => predicate(obj[key]))
+    .reduce((res, key) => ((res[key] = obj[key]), res), {})
+
+//Turns Hello World -> helloWorld
+//Taken from: https://stackoverflow.com/questions/2970525/converting-any-string-into-camel-case
+export const camelize = str => {
+  return str.replace(/(?:^\w|[A-Z]|\b\w|\s+)/g, function (match, index) {
+    if (+match === 0) return "" // or if (/\s+/.test(match)) for white spaces
+    return index === 0 ? match.toLowerCase() : match.toUpperCase()
+  })
+}
+
+//Sanitizes our filter string based on the data contained between
+//react component Fragments (ie. <Skills>{get everything between}</Skills>)
+export const sanitizeFilter = (rawBody, fragment) =>
+  rawBody
+    .substring(
+      rawBody.lastIndexOf(`<${fragment}>`) + (fragment.length + 3),
+      rawBody.lastIndexOf(`</${fragment}>`)
+    )
+    .replace(/[-.#`!*()]["]|[0-9]\./g, "") //Replace MD usual characters with empty string
```

Comment:
Ah that's a good edge case I didn't consider! I'll def slide this in with my next changes.

Reply:
@annarankin I remembered why I put the number in the regex here. It's specifically to handle that case of `1.` or `2.` instead of just base numbers. Since `X.` in the markdown spec is recognized as an ordered list.